#pYPK5

Cassette MX4 amplififcação:

 - primers 477 e 531
 - template pAG32.

Cassette 2u amplificação:

 - primers 534 e 535
 - template pYPK0

pYPK0 digerido com StuI.

In [1]:
import pydna

pYPK0 = pydna.read("pYPK0.gb")

import requests
gistid = "c5424b7ebbf553c52053"
r = requests.get('https://api.github.com/gists/'+gistid)
pAG32 = pydna.read(r.json()["files"].values().pop()["content"])

[pAG32 sequence](https://gist.github.com/BjornFJohansson/c5424b7ebbf553c52053)

In [2]:
p477, p531, p535, p534= pydna.parse(''' 
>477_bleMX4r
agtggaacgaaaactcacgttaagggattttggtcatgaggagctcgttttcga

>531_pSU1_MX4f (67-mer)
taattctcatctttgacagcttatcatcgataagtccccgccgggtc

>535_pSU0_2u_Fwd (21-mer)
taactgtcagaccaagtttac

>534_pSU0_2u_rev (27-mer)
ttatcgatgataagctgtcaaagatga''')

hygmarker = pydna.pcr(p477, p531, pAG32)

two_micron = pydna.pcr(p534, p535, pYPK0)

from Bio.Restriction import StuI

vect, stuffer = pYPK0.cut(StuI)

asm=pydna.Assembly([hygmarker, two_micron, vect])

In [3]:
asm

Assembly:
Sequences........................: [1721] [1619] [4982]
Sequences with shared homologies.: [1721] [1619] [1619] [1721] [4982] [4982]
Homology limit (bp)..............: 25
Number of overlaps...............: 9
Nodes in graph(incl. 5' & 3')....: 8
Only terminal overlaps...........: No
Circular products................: [7219] [6299]
Linear products..................: [7804] [6784] [6332] [6218] [6016] [5902] [5096] [3307] [2539] [2425] [1505] [485] [33]

In [4]:
x,y = asm.circular_products

assert len(pYPK5) == 6299

The larger recombination product x 7219 contains the 
pYPK0 vector two times in a row
the product we want is the y 6299bp

    x:
     -|1619bp_PCR_prod|1505
    |                  \/
    |                  /\
    |                  1505|pYPK0|66
    |                             \/
    |                             /\
    |                             66|pYPK0|41
    |                                      \/
    |                                      /\
    |                                      41|1721bp_PCR_pr_rc|33
    |                                                          \/
    |                                                          /\
    |                                                          33-
    |                                                             |
     -------------------------------------------------------------


    y:
     -|1619bp_PCR_prod|66
    |                  \/
    |                  /\
    |                  66|pYPK0|41
    |                           \/
    |                           /\
    |                           41|1721bp_PCR_pr_rc|33
    |                                               \/
    |                                               /\
    |                                               33-
    |                                                  |
     --------------------------------------------------

In [5]:
y.figure()

 -|1619bp_PCR_prod|66
|                  \/
|                  /\
|                  66|pYKP0|41
|                           \/
|                           /\
|                           41|1721bp_PCR_pr_rc|33
|                                               \/
|                                               /\
|                                               33-
|                                                  |
 --------------------------------------------------

In [6]:
pYPK5 = y.synced(pYPK0)

In [7]:
pYPK5.locus = "pYPK5"
pYPK5.stamp()

cSEGUID_bYXaKaxDTMFN5_B-QWoDqATVAb0_2015-07-15T15:49:51.630069

In [8]:
pYPK5.write("pYPK5.gb")

###[pYPK5](pYPK5.gb)

[DOWNLOAD](pYPK5.gb)

In [9]:
import pydna
r = pydna.read("pYPK5.gb")
r.verify_stamp()

cSEGUID_bYXaKaxDTMFN5_B-QWoDqATVAb0